proposals/0245-sharing-wifi-ssid-and-password.md

```diff
@@ -18,627 +18,1033 @@ If WiFi credentials are automatically generated by the IVI system, the password
-6. The policy table should be updated to include the below-mentioned RPCs. These RPCs should be updated in a separate `functional group`, so that, if required, the user can specifically disable WiFi credentials and status sharing RPCs with their app while allowing other RPCs. OEMs, if they require, can enable the RPC message protection feature for the below-mentioned Mobile APIs.
-7. SDL Core should check the policy table and user consent for the below-mentioned Mobile APIs. 
+### Feature initialization
 
-```xml
-<enum name="FunctionID" internal_scope="base" since="1.0">
-  .
-  .
-  .
-  <element name="GetWiFiNetworkInfoID" value="XX" hexvalue="XX" since="X.X" />
-  <element name="JoinWiFiNetworkID" value="XX" hexvalue="XX" since="X.X" />
-</enum>
+After SDL Core defines suitable network configuration and finds available feature enabled application, it has to notify HMI or mobile application.
 
-<enum name="RequestType">
-  .
-  .
-  .
-  <element name="WIFI_CONNECTION_REQ">
-    <description>This request type is sent to the mobile device when the user tries to launch an app on the IVI, which requires a secondary transport to function</description>
-  </element>
-</enum>
+If head unit should be the network host: SDL Core requests HMI sets up a WiFi network as an `OnSystemCapabilities` notification that includes `NetworkingCapabilities` and `VEHICLE` as the `preferredNetworkHost`.
 
-<struct name="DeviceInfo" since="X.X">
-  .
-  .
-  .
-  <param name="wifiConfigurationInfo" type="WiFiConfigurationInfo" mandatory="false">
-      <description>Device's available configurations around WiFi networking.</description>
-  </param>      
-</struct> 
+If the mobile device is selected to be the network host the selected app will receive the `OnSystemCapabilityUpdated` notification with the info that it was chosen to host the network.
 
-<struct name="WiFiConfigurationInfo" since="x.x">
-    <description>Describes some of the sending device's available configurations around WiFi networking. This includes the ability to automatically join a network or host one itself.</description>
-    <param name="autoJoinWiFiSupported" type="Boolean" mandatory="false">
-        <description>Can join WiFi access point created by IVI; default value is false.</description>
-    </param>
-    <param name="canHostWiFiNetwork" type="Boolean" mandatory="false">
-        <description>Can host WiFi network; default value is false.</description>
-    </param> 
-    </param> 
-</struct>
+Finally, to share the credentials of the network, it is proposed to implement the new RPC `JoinNetwork` to be sent by current networkHost.
+
+Core then sends the `TransportEventUpdate` protocol message with IP and port to connect the TCP transport to all appropriate apps. Flow is then as defined in [0141-multiple-transports.md](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0141-multiple-transports.md)

```

Comment:
Thank you for adding this language. However, more changes are needed than simply copying and pasting section 7 from the comments. Please make sure the proposal properly reflects what this section defines.

Reply:
This review point is not clear enough for me. 
After Wifi is established, SDL will notify all connected Mobiles with TransportEventUpdate according to 0141-multiple-transports. 
At this point feature initialization is finished and  and 0141-multiple-transports comes into play. I do not see additional staff that should be added to this proposal.

Reply:
I have provided a suggested change with additional information.